lib/inputs/os-release/static.ts

```diff
@@ -0,0 +1,92 @@
+import { getContentIfExists } from "../../extractor";
+import { ExtractAction, ExtractedLayers } from "../../extractor/types";
+import { streamToString } from "../../stream-utils";
+
+export { getOsReleaseActions, getOsRelease };
+
+const getOsReleaseAction: ExtractAction = {
+  actionName: "os-release",
+  fileNamePattern: "/etc/os-release",
+  callback: (stream) => streamToString(stream),
+};
+
+const getLsbReleaseAction: ExtractAction = {
+  actionName: "lsb-release",
+  fileNamePattern: "/etc/lsb-release",
+  callback: (stream) => streamToString(stream),
+};
+
+const getDebianVersionAction: ExtractAction = {
+  actionName: "debian-version",
+  fileNamePattern: "/etc/debian_version",
+  callback: (stream) => streamToString(stream),
+};
+
+const getAlpineReleaseAction: ExtractAction = {
+  actionName: "alpine-release",
+  fileNamePattern: "/etc/alpine-release",
+  callback: (stream) => streamToString(stream),
+};
+
+const getRedHatReleaseAction: ExtractAction = {
+  actionName: "redhat-release",
+  fileNamePattern: "/etc/redhat-release",
+  callback: (stream) => streamToString(stream),
+};
+
+const getOracleReleaseAction: ExtractAction = {
+  actionName: "oracle-release",
+  fileNamePattern: "/etc/oracle-release",
+  callback: (stream) => streamToString(stream),
+};
+
+const getOsReleaseActions: ExtractAction[] = [
+  getOsReleaseAction,
+  getLsbReleaseAction,
+  getDebianVersionAction,
+  getAlpineReleaseAction,
+  getRedHatReleaseAction,
+  getOracleReleaseAction,
+];
+
+function getOsRelease(extractedLayers: ExtractedLayers): string {
+  const osRelease = getContentIfExists(extractedLayers, getOsReleaseAction);
+  return osRelease !== undefined ? (osRelease as string) : "";
+}
+
+export function getLsbRelease(extractedLayers: ExtractedLayers): string {
+  const lsbRelease = getContentIfExists(extractedLayers, getLsbReleaseAction);
+  return lsbRelease !== undefined ? (lsbRelease as string) : "";
+}
+
+export function getDebianVersion(extractedLayers: ExtractedLayers): string {
+  const debianVersion = getContentIfExists(
+    extractedLayers,
+    getDebianVersionAction,
+  );
+  return debianVersion !== undefined ? (debianVersion as string) : "";
+}
+
+export function getAlpineRelease(extractedLayers: ExtractedLayers): string {
```

Comment:
all the functions here have the same functionality and receive the same params, I think they only differ by the extract action. can't we have only one function e.g. `getDistributionRelease` and pass the distribution type as a param?

Reply:
It's a very good improvement and it should also affect the `docker` side of this where it also does a chain of calls like these ones. But in order to abstract this I need another layer of indirection, something like a map that can process the input given the DistroType... I'll think about it and see if I can improve it!

Reply:
Done!